### PR TITLE
git-bug: tweaks; git-bug-migration: init at 0.3.4

### PR DESCRIPTION
**Pull request by DeeUnderscore**
###### Description of changes
git-bug changelog: https://github.com/MichaelMure/git-bug/releases/tag/v0.8.0

git-bug includes backwards incompatible changes. [git-bug-migration](https://github.com/MichaelMure/git-bug-migration) is a separate tool for upgrading the bug database in existing repositories, so I'm packaging it here as well. Included is also a changelog hint about the migration tool. 

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by DeeUnderscore** [comment]
Retargeted changelog to 23.05

**Comment by royneary** [comment]
Thanks! The commit hash for tag ``v0.8.0`` seems to have changed. I get 

```
error: builder for '/nix/store/g5ijspgbk769dii2ci55lx5pzvajc0xf-source.drv' failed with exit code 1;
       last 8 log lines:
       >
       > trying https://github.com/MichaelMure/git-bug/archive/f7b6421bf2ca760fcd61d17ea320e8bbf331783e.tar.gz
       >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
       >                                  Dload  Upload   Total   Spent    Left  Speed
       >   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       >   0    14    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       > curl: (22) The requested URL returned error: 404
       > error: cannot download source from any mirror
```

**Comment by DeeUnderscore** [comment]
It actually hasn't changed, I just accidentally used a git-bug-migration ref instead of git-bug. Thanks for catching it.

**Review by royneary — APPROVED** [review]

**Review by royneary — APPROVED** [review]

**Comment by DeeUnderscore** [comment]
Rebased on master.

#205328 preempted the version update, but I think disabling the unneeded builds and installing fish completions is still useful.

git-bug-migration is still a new package.

**Review by royneary — APPROVED** [review]

**Comment by DeeUnderscore** [comment]
Rebased to resolve merge conflicts.

File: nixos/doc/manual/release-notes/rl-2305.section.md [hunk 1]
```diff
@@ -42,6 +42,8 @@ In addition to numerous new and upgraded packages, this release has the followin
 
 - `podman` now uses the `netavark` network stack. Users will need to delete all of their local containers, images, volumes, etc, by running `podman system reset --force` once before upgrading their systems.
 
+- `git-bug` has been updated to version 0.8.0, which includes backwards incompatible changes. The `git-bug-migration` package can be used to upgrade existing repositories.
```
Comment (SuperSandro2000) [inline nixos/doc/manual/release-notes/rl-2305.section.md:45]: Rather unusual to mention single packages.

```suggestion
- `git-bug` has been updated to at least version 0.8.0, which includes backwards incompatible changes. The `git-bug-migration` package can be used to upgrade existing repositories.
```

File: pkgs/applications/version-management/git-bug/default.nix [hunk 2]
```diff
@@ -1,9 +1,9 @@
-{ lib, buildGoModule, fetchFromGitHub }:
+{ lib, buildGoModule, fetchFromGitHub, installShellFiles }:
 
 buildGoModule rec {
   pname = "git-bug";
-  version = "0.8.0"; # the `rev` below pins the version of the source to get
-  rev = "v0.8.0";
+  version = "0.8.0"; # explicit commit hash is passed in ldflags, needed for version info
+  rev = "a3fa445a9c76631c4cd16f93e1c1c68a954adef7";
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/git-bug/default.nix:6]:
```suggestion
  version = "0.8.0";
```
we don't need that and makes auto updates easier

File: pkgs/applications/version-management/git-bug/default.nix [hunk 3]
```diff
@@ -14,29 +14,31 @@ buildGoModule rec {
...
   vendorSha256 = "sha256-32kNDoBE50Jx1Ef9YwhDk7nd3CaTSnHPlu7PgWPUGfE=";
 
+  nativeBuildInputs = [ installShellFiles ];
+
   doCheck = false;
 
+  excludedPackages = [ "doc" "misc" ];
+
   ldflags = [
     "-X github.com/MichaelMure/git-bug/commands.GitCommit=${rev}"
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/git-bug/default.nix:24]:
```suggestion
    "-X github.com/MichaelMure/git-bug/commands.GitCommit=v${version}"
```

File: pkgs/applications/version-management/git-bug/default.nix (hunk 2 above)
Comment (DeeUnderscore) [reply pkgs/applications/version-management/git-bug/default.nix]: There is a `git-bug version --commit` command which normally outputs the whole hash, but I guess that's a small breakage if someone is relying on that to be a hash and not any ref.

**Pull request merged**